### DISCUSSION TITLE
Support for NextJS 14

### DISCUSSION THREAD

**Discussion by julianriise** [chakra-ui/chakra-ui#8863] [General]
### Description

I'm unable to compile my NextJS 14 + ChakraUI project because of a long list of issues. Am I correct to assume that Chakra don't support NextJS 14? And will there be support for it in near future?

### Problem Statement/Justification

As a developer I'm unable to use Chakra in the current state.

### Proposed Solution or API

Please update the [NextJS page](https://chakra-ui.com/getting-started/nextjs-guide) on Chakra website with more information.

<img width="786" alt="Screenshot 2023-10-29 at 10 34 38" src="https://github.com/chakra-ui/chakra-ui/assets/77626314/11c70409-f8c9-4907-a5fc-bcd1ffad0297">

**Comment by mymai91** [comment]
@julianriise 

I can run chakraUI with NextJs 14.0.2


Here is my package.json you can refer

```
{
  "name": "app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "PORT=8000 next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@chakra-ui/next-js": "^2.2.0",
    "@chakra-ui/react": "^2.8.2",
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "framer-motion": "^10.16.4",
    "next": "14.0.2",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "14.0.2",
    "typescript": "^5"
  }
}
```

Modify `layout.tsx`

```
import { ChakraProvider } from "@chakra-ui/react"
import type { Metadata } from "next"
import { Inter } from "next/font/google"
import "./globals.css"
import theme from "./theme"

const inter = Inter({ subsets: ["latin"] })

export const metadata: Metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
}

export default function RootLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html lang="en">
      <body className={inter.className}>
        <ChakraProvider theme={theme} cssVarsRoot="body">
          {children}
        </ChakraProvider>
      </body>
    </html>
  )
}
```

**Comment by HMaroba** [comment]
@mymai91   But on production it fails on my side on vercel but local it works fine

**Comment by gstadtler** [comment]
> @julianriise
> 
> I can run chakraUI with NextJs 14.0.2
> 
> Here is my package.json you can refer
> 
> ```
> {
>   "name": "app",
>   "version": "0.1.0",
>   "private": true,
>   "scripts": {
>     "dev": "PORT=8000 next dev",
>     "build": "next build",
>     "start": "next start",
>     "lint": "next lint"
>   },
>   "dependencies": {
>     "@chakra-ui/next-js": "^2.2.0",
>     "@chakra-ui/react": "^2.8.2",
>     "@emotion/react": "^11.11.1",
>     "@emotion/styled": "^11.11.0",
>     "framer-motion": "^10.16.4",
>     "next": "14.0.2",
>     "react": "^18",
>     "react-dom": "^18"
>   },
>   "devDependencies": {
>     "@types/node": "^20",
>     "@types/react": "^18",
>     "@types/react-dom": "^18",
>     "eslint": "^8",
>     "eslint-config-next": "14.0.2",
>     "typescript": "^5"
>   }
> }
> ```
> 
> Modify `layout.tsx`
> 
> ```
> import { ChakraProvider } from "@chakra-ui/react"
> import type { Metadata } from "next"
> import { Inter } from "next/font/google"
> import "./globals.css"
> import theme from "./theme"
> 
> const inter = Inter({ subsets: ["latin"] })
> 
> export const metadata: Metadata = {
>   title: "Create Next App",
>   description: "Generated by create next app",
> }
> 
> export default function RootLayout({
>   children,
> }: {
>   children: React.ReactNode
> }) {
>   return (
>     <html lang="en">
>       <body className={inter.className}>
>         <ChakraProvider theme={theme} cssVarsRoot="body">
>           {children}
>         </ChakraProvider>
>       </body>
>     </html>
>   )
> }
> ```

My package.json looks just like yours and still I'm getting this error that I can't get through
![image](https://github.com/chakra-ui/chakra-ui/assets/50388072/c5bdf819-6530-48cc-b8d6-b5f680ae4ed7)

I have also configured the chakra providers on my layout.tsx file and it didn't make any difference. Any thoughts?

**Comment by FaezehDqnn** [comment]
> > @julianriise
> > I can run chakraUI with NextJs 14.0.2
> > Here is my package.json you can refer
> > ```
> > {
> >   "name": "app",
> >   "version": "0.1.0",
> >   "private": true,
> >   "scripts": {
> >     "dev": "PORT=8000 next dev",
> >     "build": "next build",
> >     "start": "next start",
> >     "lint": "next lint"
> >   },
> >   "dependencies": {
> >     "@chakra-ui/next-js": "^2.2.0",
> >     "@chakra-ui/react": "^2.8.2",
> >     "@emotion/react": "^11.11.1",
> >     "@emotion/styled": "^11.11.0",
> >     "framer-motion": "^10.16.4",
> >     "next": "14.0.2",
> >     "react": "^18",
> >     "react-dom": "^18"
> >   },
> >   "devDependencies": {
> >     "@types/node": "^20",
> >     "@types/react": "^18",
> >     "@types/react-dom": "^18",
> >     "eslint": "^8",
> >     "eslint-config-next": "14.0.2",
> >     "typescript": "^5"
> >   }
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Modify `layout.tsx`
> > ```
> > import { ChakraProvider } from "@chakra-ui/react"
> > import type { Metadata } from "next"
> > import { Inter } from "next/font/google"
> > import "./globals.css"
> > import theme from "./theme"
> > 
> > const inter = Inter({ subsets: ["latin"] })
> > 
> > export const metadata: Metadata = {
> >   title: "Create Next App",
> >   description: "Generated by create next app",
> > }
> > 
> > export default function RootLayout({
> >   children,
> > }: {
> >   children: React.ReactNode
> > }) {
> >   return (
> >     <html lang="en">
> >       <body className={inter.className}>
> >         <ChakraProvider theme={theme} cssVarsRoot="body">
> >           {children}
> >         </ChakraProvider>
> >       </body>
> >     </html>
> >   )
> > }
> > ```
> 
> My package.json looks just like yours and still I'm getting this error that I can't get through ![image](https://private-user-images.githubusercontent.com/50388072/284776305-c5bdf819-6530-48cc-b8d6-b5f680ae4ed7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMyMzgxNjAsIm5iZiI6MTcwMzIzNzg2MCwicGF0aCI6Ii81MDM4ODA3Mi8yODQ3NzYzMDUtYzViZGY4MTktNjUzMC00OGNjLWI4ZDYtYjVmNjgwYWU0ZWQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIyVDA5Mzc0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE5MTRkNzEzZWMwZTA2ZDYzYThkNzlmNTczMWU2NjE1N2NjY2U0ZTRlODU3YjQ2OWZmM2M3NmEwMmVmODVmNjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yCzuEMIokKXY9WET4SCcughyA7NhFNSblKLZ6Ur9yJY)
> 
> I have also configured the chakra providers on my layout.tsx file and it didn't make any difference. Any thoughts?

I have the same issue and couldn't find any useful solutions :(

**Comment by Omikorin** [comment]
@FaezehDqnn could you provide a minimal reproduction example? That's necessary to help

**Comment by runvnc** [comment]
I've been going through a lot of stuff.. ran into issues with useBreakpointValue and getting useSession/SessionProvider to work. The problems started when I tried to integrate next-auth. I think I have to give up and switch to some alternative to Chakra-UI.  If there is something that exists and works with Next.js 14.

EDIT: I think I got past the major problem I was having. It might have been caused by trying to use the App Router and Pages at the same time. So I don't think I have to abandon this :)

**Comment by aaarichter** [comment]
another problem is with the chakra-ui module - 
when importing an icon with like `import { ArrowForwardIcon } from '@chakra-ui/icons';`

NextJS14 shows this error message

> ./node_modules/@chakra-ui/icons/dist/index.mjs
Error: It's currently unsupported to use "export *" in a client boundary. Please use named exports instead.

its because of 
```
export * from '@chakra-ui/icon';
```

**Comment by aaarichter** [comment]
also, it is not clear how to deal with the image component

`import {Image} from "@chakra-ui/next-js";` seems to have an image export wrapping nextjs `Image` but it is 
- not documented on https://chakra-ui.com/getting-started/nextjs-app-guide
- and fails when using it:

> Unhandled Runtime Error
> Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) > but got: object.
> 
> Check the render method of `Styled(Component)`.

**Comment by Fecoded** [comment]
Is there a plan to release support for Nextjs 14 in the future?

**Comment by tomdoeslinux** [comment]
I'm getting a ton of bugs with Next.JS 14 and ChakraUI. So much so that I've had to resort to traditional CSS.

**Comment by emyjamalian** [comment]
> @julianriise
> 
> I can run chakraUI with NextJs 14.0.2
> 
> Here is my package.json you can refer
> 
> ```
> {
>   "name": "app",
>   "version": "0.1.0",
>   "private": true,
>   "scripts": {
>     "dev": "PORT=8000 next dev",
>     "build": "next build",
>     "start": "next start",
>     "lint": "next lint"
>   },
>   "dependencies": {
>     "@chakra-ui/next-js": "^2.2.0",
>     "@chakra-ui/react": "^2.8.2",
>     "@emotion/react": "^11.11.1",
>     "@emotion/styled": "^11.11.0",
>     "framer-motion": "^10.16.4",
>     "next": "14.0.2",
>     "react": "^18",
>     "react-dom": "^18"
>   },
>   "devDependencies": {
>     "@types/node": "^20",
>     "@types/react": "^18",
>     "@types/react-dom": "^18",
>     "eslint": "^8",
>     "eslint-config-next": "14.0.2",
>     "typescript": "^5"
>   }
> }
> ```
> 
> Modify `layout.tsx`
> 
> ```
> import { ChakraProvider } from "@chakra-ui/react"
> import type { Metadata } from "next"
> import { Inter } from "next/font/google"
> import "./globals.css"
> import theme from "./theme"
> 
> const inter = Inter({ subsets: ["latin"] })
> 
> export const metadata: Metadata = {
>   title: "Create Next App",
>   description: "Generated by create next app",
> }
> 
> export default function RootLayout({
>   children,
> }: {
>   children: React.ReactNode
> }) {
>   return (
>     <html lang="en">
>       <body className={inter.className}>
>         <ChakraProvider theme={theme} cssVarsRoot="body">
>           {children}
>         </ChakraProvider>
>       </body>
>     </html>
>   )
> }
> ```

You solved my problem. thank you so much. My issue was where I was using <ChakraProvider>

**Comment by rodacode** [comment]
Guys this is still happening why is closed? Same errors, I tried to add the provider in the layout.tsx and still happening with next version 14.1.2

**Comment by emyjamalian** [comment]
@rodacode Could you please share your layout? I've been dealing with this error a lot now. I hope I can help :)

**Comment by sadi304** [comment]
> Is there a plan to release support for Nextjs 14 in the future?

Humbly asking, any roadmap/plans for RSC support?

**Comment by bodadotsh** [comment]
@sadi304 Chakra v3 (in active development) will support RSC: https://twitter.com/thesegunadebayo/status/1773403593086844939

**Comment by averypelle** [comment]
@segunadebayo thank you for your work on this repo. what is the timeline for the release of these changes (as part of the v3 release)? Proper support for NextJS 14 is essential for me to keep using this library in production, and would appreciate an update. I would also be happy to contribute.

**Comment by kevin-4-dev** [comment]
it works for me  https://v2.chakra-ui.com/getting-started/nextjs-app-guide#setup-provider
